### PR TITLE
Use openssl implementation of random, md5 and hmac-sha1.

### PR DESCRIPTION
**Pull request by vassjozsef**
Please see https://github.com/vassjozsef/libnice-openssl.

### CODE REVIEW

**Review by adill — APPROVED** [review]
lgtm 👍

it'd be nice to figure out how to integrate [your test suite](https://github.com/vassjozsef/libnice-openssl) in-repo (we tend towards monorepo hero), though it's certainly not critical.

File: stun/stunhmac-openssl.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  HMAC_CTX_cleanup(&handle);
+  HMAC_cleanup(&handle);
+}
+
+static const uint8_t *priv_trim_var(const uint8_t *var, size_t *var_len)
+{
+  const uint8_t *ptr = var;
+
+  while (*ptr == '"') {
```
Comment (snowyote) [inline stun/stunhmac-openssl.c:53]: not sure if this string is zero-terminated, but i'd like to see a `*var_len > 0` condition regardless

File: stun/stunhmac-openssl.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    ptr++;
+    (*var_len)--;
+  }
+  while(ptr[*var_len-1] == '"' ||
+      ptr[*var_len-1] == 0) {
```
Comment (snowyote) [inline stun/stunhmac-openssl.c:58]: Similarly, we should break here if we ever get down to `*var_len == 0`.  I don't know that this condition is reachable unless we're called with `*var_len == 0` initially, but it def makes me more comfortable

File: stun/stunhmac-openssl.c [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    TRY(HMAC_Update(&handle, pad_char, pad_size));
+  }
+
+  unsigned int shalen;
+  HMAC_Final(&handle, sha, &shalen);
+  assert(shalen == 20);
+
+  HMAC_CTX_cleanup(&handle);
+  HMAC_cleanup(&handle);
```
Comment (snowyote) [inline stun/stunhmac-openssl.c:46]: `#undef TRY` for hygeine?

**Review by snowyote — COMMENTED** [review]
you beat me by a minute :D

File: stun/stunhmac-openssl.c (hunk 3 above)
Comment (vassjozsef) [reply stun/stunhmac-openssl.c:46]: This was an omission, fixed.

**Comment by vassjozsef** [comment]
I addressed these comments in https://github.com/discordapp/libnice/pull/4.

**Pull request merged**